workspaces/ui-v2/src/components/navigation/TopNavigation.tsx

```diff
@@ -11,13 +11,13 @@ import { Container } from '@material-ui/core';
 import SubjectIcon from '@material-ui/icons/Subject';
 import { NavButton } from './NavButton';
 import ChangeHistoryIcon from '@material-ui/icons/ChangeHistory';
-import { useApiName } from '<src>/hooks/useApiNameHook';
 import { useDiffReviewPageLink, useDocumentationPageLink } from './Routes';
 import { useAppConfig } from '<src>/contexts/config/AppConfiguration';
+import { useAppSelector } from '<src>/store';
 
 export function TopNavigation(props: { AccessoryNavigation: any }) {
   const classes = useStyles();
-  const apiName = useApiName();
+  const apiName = useAppSelector((state) => state.metadata.data?.apiName || '');
```

Comment:
When we come to render this component, the appSelector should always return the apiName (given we are rendering loading + error states higher in the tree) - typescript cannot infer this - and there are cases where we could forget to handle loading + error states (if we refactor, etc)

My thoughts here are to:
- Create a selector that checks and throws an error if `data` is undefined - this way we have type safety and also have quicker developer feedback if we haven't explicitly handled error states

Thoughts? Any alternative (rather than having to handle these with casting to some default value)?

Reply:
I think it's fine for this not to be safe, as failure doesn't have any major impact on the rest of the app's flow, it's only being used in a single spot so it's not too hard to get right. If in QA or through user testing this pops up as being missing sometimes, we can always come back and address this (properly with the separate selector as you've mentioned).

Reply:
I think the selectors are the best long term solution - but for now this should be fine. There are cases where we have the same pattern where we'll handle loading states higher in the tree, and will want the assumption we've already loaded the data / handled errors (loading endpoint data, etc).